PumaGrasshopper/AttributeForm.cs

```diff
@@ -12,25 +12,158 @@ namespace PumaGrasshopper
+                    }
+                case Annotations.AnnotationArgumentType.AAT_INT:
+                    {
+                        defaultValueMap.GetInteger(attributeName, out int val);
+                        return val.ToString();
+                    }
+                case Annotations.AnnotationArgumentType.AAT_BOOL_ARRAY:
+                    {
+                        defaultValueMap.GetBoolArray(attributeName, out bool[] vals);
+                        return vals.Aggregate("", (left, right) => left + right + ";");
+                    }
+                case Annotations.AnnotationArgumentType.AAT_FLOAT_ARRAY:
+                    {
+                        defaultValueMap.GetDoubleArray(attributeName, out double[] vals);
+                        return vals.Aggregate("", (left, right) => left + right + ";");
+                    }
+                case Annotations.AnnotationArgumentType.AAT_STR_ARRAY:
+                    {
+                        defaultValueMap.GetStringArray(attributeName, out string[] vals);
+                        return vals.Aggregate("", (left, right) => left + right + ";");
+                    }
+                case Annotations.AnnotationArgumentType.AAT_INT_ARRAY:
+                    {
+                        defaultValueMap.GetIntegerArray(attributeName, out int[] vals);
+                        return vals.Aggregate("", (left, right) => left + right + ";");
+                    }
+                default:
+                    return "No default value";
+            }
+        }
+
+        private List<ListViewItem> getAttributesListViewItems(List<RuleAttribute> ruleAttributes)
+        {
+            List<ListViewItem> listViewItems = ruleAttributes.ConvertAll(
+                attribute => {
+                    return new ListViewItem(new string[] {
+                            attribute.mNickname,
+                            GetTypeString(attribute.mAttribType),
+                            GetDefaultValueString(attribute.mFullName, attribute.mAttribType)
+                        },
+                        group: Groups.Find(group => group.Header == attribute.mGroup));
+                    }
+                );
+
+            return listViewItems;
+        }
+
+        private ListViewGroup[] getAllGroups(List<RuleAttribute> attributes)
+        {
+            var groupNames = attributes.ConvertAll(attr => attr.mGroup);
+
+            HashSet<string> groups = new HashSet<string>();
+            groupNames.ForEach(groupName => groups.Add(groupName));
+
+            List<ListViewGroup> groupListView = new List<ListViewGroup>();
+            groups.ToList().ForEach((groupName) => groupListView.Add(new ListViewGroup(groupName)));
+
+            return groupListView.ToArray();
         }
 
         private void AttributeForm_Load(object sender, EventArgs e)
         {
             SetDesktopLocation(InitialLocation.X, InitialLocation.Y);
-            AttributeComboBox.SelectedIndex = 0;

```

Comment:
I guess this is already the default?

Reply:
This was useful before because we used a combo box, which has by default an element selected. I removed the combo box so this was also removed. Also, the list view has no element selected by default.